src/ApisHandler.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+
+import { each, isArray } from "lodash";
+
+import { defaultConfig } from "./defaultConfig";
+
+export class ApisHandler {
+  constructor() {
+    this.reset();
+  }
+
+  _ensureArray(tags) {
+    if (!tags) {
+      return [];
+    }
+    return isArray(tags) ? tags : [tags];
+  }
+
+  _setBaseConfig(config) {
+    this._baseConfig = { ...this._baseConfig, ...config };
+    each(this._all, api => {
+      api.config(this._baseConfig);
+    });
+  }
+
+  _setTagConfig(config, tag) {
+    this._configs[tag] = { ...this._configs[tag], ...config };
+    each(this._tagged[tag], api => {
+      api.config(this._configs[tag]);
+    });
+  }
+
+  _setBaseHeaders(headers) {
+    this._baseHeaders = headers;
+    each(this._all, api => {
+      api.setHeaders(this._baseHeaders);
+    });
+  }
+
+  _setTagHeaders(headers, tag) {
+    this._headers[tag] = headers;
+    each(this._tagged[tag], api => {
+      api.setHeaders(this._headers[tag]);
+    });
+  }
+
+  _addBaseHeaders(headers) {
+    this._baseAddedHeaders = { ...this._baseAddedHeaders, ...headers };
+    each(this._all, api => {
+      api.addHeaders(this._baseAddedHeaders);
+    });
+  }
+
+  _addTagHeaders(headers, tag) {
+    this._addedHeaders[tag] = { ...this._addedHeaders[tag], ...headers };
+    each(this._tagged[tag], api => {
+      api.addHeaders(this._addedHeaders[tag]);
+    });
+  }
+
+  _cleanAll() {
+    each(this._all, api => {
+      api.clean();
+    });
+  }
+
+  _clean(tag) {
+    each(this._tagged[tag], api => {
+      api.clean();
+    });
+  }
+
+  register(api, tags) {
+    this._all.push(api);
+    this._ensureArray(tags).forEach(tag => {
+      this._tagged[tag] = this._tagged[tag] || [];
+      this._tagged[tag].push(api);
+    });
+  }
+
+  getConfig(customConfig) {
+    const clonedCustomConfig = { ...customConfig };
+    delete clonedCustomConfig.tags;
+    let totalConfig = { ...defaultConfig, ...this._baseConfig, ...clonedCustomConfig };
+    this._ensureArray(customConfig.tags).forEach(tag => {
+      totalConfig = {
+        ...totalConfig,
+        ...this._configs[tag]
+      };
+    });
+    return totalConfig;
+  }
+
+  getHeaders(tags) {
+    let headers = { ...this._baseHeaders, ...this._baseAddedHeaders };
+    this._ensureArray(tags).forEach(tag => {
+      headers = {
```

Comment:
It's desirable to mention (in docs) that in case two tags have common properties the order is important and the last defined tag has priority.

Reply:
Good point, thank you. I have added an apis configuration example which includes a section about hipotetical conflicts between tags values, and tests ensuring that the behavior in case of conflict is as expected. You can check the changes in the commit:

https://github.com/XbyOrange/mercury-api/pull/8/commits/a4bbd844b4482fca7e87e811197dc8c72ae49316